subcommand/server-acl-init/command.go

```diff
@@ -0,0 +1,304 @@
+		return 1
+	}
+
+	level := hclog.LevelFromString(c.flagLogLevel)
+	if level == hclog.NoLevel {
+		c.UI.Error(fmt.Sprintf("Unknown log level: %s", c.flagLogLevel))
+		return 1
+	}
+
+	logger := hclog.New(&hclog.LoggerOptions{
+		Level:  level,
+		Output: os.Stderr,
+	})
+
+	// Use the client to get statefulset pods
+	labelSelector := fmt.Sprintf("component=server, app=consul, release=%s", c.flagReleaseName)
+
+	serverPods, err := clientset.CoreV1().Pods(c.flagNamespace).List(metav1.ListOptions{LabelSelector: labelSelector})
+	if err != nil {
+		logger.Error(err.Error())
+		return 1
+	}
+
+	if len(serverPods.Items) == 0 {
+		logger.Error("No pods were found")
+		return 1
+	}
+
+	// Pull the addresses out of each pod
+	var podAddresses []string
+	for _, pod := range serverPods.Items {
+		address := fmt.Sprintf("%s:8500", pod.Status.PodIP)
+		if pod.Status.PodIP != "" {
+			podAddresses = append(podAddresses, address)
+			logger.Info(address)
+		}
+	}
+
+	if len(podAddresses) < c.flagReplicas {
+		logger.Error(fmt.Sprintf("Not enough pod addresses were found: %d", len(podAddresses)))
+		return 1
+	}
+
+	// Pick the first pod to connect to for bootstrapping & set up connection
+	consulConfig := api.Config{
+		Address: podAddresses[0],
+		Scheme:  "http",
+	}
+
+	consulClient, err := api.NewClient(&consulConfig)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error connecting to Consul agent: %s", err))
+		return 1
+	}
+
+	// Bootstrap the ACLs
+	bootstrapToken, _, err := consulClient.ACL().Bootstrap()
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error bootstrapping Consul agent: %s", err))
+		return 1
+	}
+
+	// Write bootstrap token to a Kubernetes secret
+	_, err = clientset.CoreV1().Secrets(c.flagNamespace).Create(&apiv1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-consul-bootstrap-acl-token", c.flagReleaseName),
+		},
+		StringData: map[string]string{
+			"token": bootstrapToken.SecretID,
+		},
+	})
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error creating bootstrap token secret: %s", err))
+		return 1
+	}
+
+	// Create agent policy
+	policy := `node_prefix "" {
+   policy = "write"
+}
+service_prefix "" {
+   policy = "read"
+}`
+
+	agentPolicy := api.ACLPolicy{
+		Name:        "agent-token",
+		Description: "Agent Token Policy",
+		Rules:       policy,
+	}
+
+	aclPolicy, _, err := consulClient.ACL().PolicyCreate(&agentPolicy, &api.WriteOptions{Token: bootstrapToken.SecretID})
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error creating agent policy: %s", err))
+		return 1
+	}
+
+	// Create agent token for each agent
+	var serverTokens []api.ACLToken
+	token := api.ACLToken{
+		Description: "Server Agent Token",
+		Policies:    []*api.ACLTokenPolicyLink{&api.ACLTokenPolicyLink{Name: aclPolicy.Name}},
+	}
+
+	for i := 0; i < len(podAddresses); i++ {
+		newToken, _, err := consulClient.ACL().TokenCreate(&token, &api.WriteOptions{Token: bootstrapToken.SecretID})
```

Comment:
You may wish to customize the `token.Description` field here to indicate which pod each is intended for

Reply:
Added the server pod name in with each token.